File: python/recursive-algos/binary_search.py
def binary_search(haystack, needle, start, end):
    # Base case  
    if start > end:
        return -1
    # Finding middle value
    mid = (start + end) // 2
    # Rite of passage
    if haystack[mid] == needle:
        return mid
    
    # Recursive case  
    return binary_search(haystack, needle, mid+1, end) if needle > haystack[mid] else binary_search(haystack, needle, start, mid-1)
    
my_list = [1,2,6,7,8,9,10]

print(binary_search(my_list, 7, 0, len(my_list)-1))

File: javascript/recursive-algos/binarySearch.js
let binarySearch = (haystack, needle, start, end) => {
    if (haystack.length == 0 || start > end) return -1 

    let mid = Math.floor((start + end) / 2)

    if (haystack[mid] == needle) return mid

    else if (needle > haystack[mid]) {
        return binarySearch(haystack, needle, mid + 1, end)
    }

    else if (needle < haystack[mid]) {
        return binarySearch(haystack, needle, start, mid - 1)
    }
}

let list = [1, 2, 4, 5, 6, 8, 10, 12, 15]
console.log(binarySearch(list, 7, 0, list.length - 1))